client/gutenberg/extensions/map-block/edit.js

```diff
@@ -45,10 +66,46 @@ class MapEdit extends Component {
 		// Allow one cycle for alignment change to take effect
 		setTimeout( this.mapRef.current.sizeMap, 0 );
 	};
+	updateAPIKeyControl = value => {
+		this.setState( { apiKeyControl: value }, this.debouncedUpdateAPIKey );
+	};
+	updateAPIKey = () => {
+		const { noticeOperations } = this.props;
+		const { apiKeyControl } = this.state;
+		noticeOperations.removeAllNotices();
+		apiKeyControl && this.apiCall( apiKeyControl, 'POST' );
+	};
+	removeAPIKey = () => {
+		this.apiCall( null, 'DELETE' );
+	};
+	apiCall( api_key = null, method = 'GET' ) {
+		const { noticeOperations } = this.props;
+		const url = '/wp-json/jetpack/v4/api-key/googlemaps';
+		const fetch = api_key ? { url, method, data: { api_key } } : { url, method };
+		apiFetch( fetch )
+			.then( result => {
+				noticeOperations.removeAllNotices();
+				this.setState( {
+					apiState: result.api_key ? API_STATE_SUCCESS : API_STATE_FAILURE,
+					api_key: result.api_key,
+					apiKeyControl: result.api_key,
+				} );
+			} )
+			.catch( result => {
+				noticeOperations.removeAllNotices();
+				noticeOperations.createErrorNotice( result.message );
+				this.setState( {
+					apiState: API_STATE_FAILURE,
+				} );
+			} );
```

Comment:
this is fine here I think given the code that's actually executing but a quick note on `.then()` is that we can pass two parameters or one. with two parameters the second is actually the `catch` but it differs from `.then().catch()` because when both are supplied to `then` then the catch only catches from the original promise.

I'm sure that's confusing, so here is some code

```js
Promise
	.resolve()
	.then( () => { throw 'a' }, () => { throw 'b' } )
	// threw 'a'

Promise
	.reject()
	.then( () => { throw 'a' }, () => { throw 'b' } )
	// threw 'b'

Promise
	.resolve()
	.then( () => { throw 'a' } )
	.catch( () => { throw 'b' } )
	// threw 'b'

Promise
	.reject()
	.then( () => { throw 'a' } )
	.catch( () => { throw 'b' } )
	// threw 'b'
```

notice how in the last two cases `'b'` threw in both the resolution and rejection part? that's because _the exception which happened inside of the `then()` cascaded into the following `catch()` chain_. it _looks_ like the `catch()` is there to grab the failure from the original promise but actually it's grabbing _both_ a failure in the original promise _and_ a failure in the chained `then()`

hope that's useful! (@blowery is good at finding bugs from this type of thing and fixing them)